### PR TITLE
Correct out-of-date docs for StaticHashTable (redo of #28199)

### PR DESCRIPTION
**Pull request by frreiss**
The API documentation for `tf.lookup.StaticHashTable` hasn't been updated to reflect recent code changes. This PR updates the docstring for both the V1 and V2 entry points to that class with code snippets that work on the latest master branch. I've included separate example code for both eager mode and graph mode.

### CODE REVIEW

**Comment by gbaned** [comment]
@MarkDaoust Could you PTAL and approve.

**Comment by gbaned** [comment]
@MarkDaoust can you please review this PR ?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @MarkDaoust, @rohan100jain: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/python/ops/lookup_ops.py [hunk 1]
```diff
@@ -239,16 +239,18 @@ def initializer(self):
 
 @tf_export("lookup.StaticHashTable", v1=[])
 class StaticHashTable(InitializableLookupTableBase):
-  """A generic hash table implementation.
+  """A generic hash table implementation. The table is immutable once
```
Comment (alextp) [inline tensorflow/python/ops/lookup_ops.py]: Linter will complain that the summary needs to be one line ending with a period.

**Review by alextp — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/lookup_ops.py (hunk 1 above)
Comment (frreiss) [reply tensorflow/python/ops/lookup_ops.py]: Thanks for the heads-up! I've pushed a fix to my branch.

**Review by alextp — APPROVED** [review]

**Pull request merged**